web/src/components/betaBar/BetaBar.test.tsx

```diff
@@ -0,0 +1,178 @@
+      render(<BetaBar />);
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.queryByText(Config.betaBar.betaModalFeedbackButtonText)).not.toBeInTheDocument();
+    });
+
+    it("links to correct feedback form when completed onboarding and owns a business", () => {
+      useMockUserData({
+        formProgress: "COMPLETED",
+        profileData: generateProfileData({ hasExistingBusiness: true }),
+      });
+      render(<BetaBar />);
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.getByText(Config.betaBar.betaModalFeedbackButtonText)).toBeInTheDocument();
+      expect(screen.getByTestId("feedback-link").getAttribute("href")).toEqual(
+        Config.betaBar.betaFormLinkOwning
+      );
+    });
+
+    it("links to correct feedback form when completed onboarding and starting a business", () => {
+      useMockUserData({
+        formProgress: "COMPLETED",
+        profileData: generateProfileData({ hasExistingBusiness: false }),
+      });
+      render(<BetaBar />);
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.getByText(Config.betaBar.betaModalFeedbackButtonText)).toBeInTheDocument();
+      expect(screen.getByTestId("feedback-link").getAttribute("href")).toEqual(
+        Config.betaBar.betaFormLinkStarting
+      );
+    });
+  });
+
+  describe("request a feature", () => {
+    it("does not show feature request button when userData is undefined", () => {
+      useUndefinedUserData();
+      render(<BetaBar />);
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.queryByText(Config.betaBar.betaModalFeatureRequestButtonText)).not.toBeInTheDocument();
+    });
+
+    it("clears feature request input when modal is closed and is revisted", () => {
+      useMockUserData({
+        formProgress: "COMPLETED",
+        profileData: generateProfileData({ hasExistingBusiness: false }),
+      });
+      render(<BetaBar />);
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText)).toBeInTheDocument();
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText));
+      fireEvent.change(screen.getByLabelText("Feature request"), { target: { value: "random text" } });
+      expect((screen.getByLabelText("Feature request") as HTMLInputElement).value).toBe("random text");
+
+      fireEvent.click(screen.getByLabelText("close"));
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText));
+      expect((screen.getByLabelText("Feature request") as HTMLInputElement).value).toBe("");
+      expect(screen.queryByText(Config.betaBar.featureRequestModalInlineErrorText)).not.toBeInTheDocument();
+    });
```

Comment:
having expect statements scattered through this test makes me think it's actually testing several different things. I think these should be broken out into independent tests so that it is more clear and readable what logic is at play here.

- it only shows the Feature Request button when form progress is completed
- it does not show the Feature Request button when form progress is NOT completed
- it clears feature request input when modal is closed and is revisited